components/AddressPill.tsx

```diff
@@ -1,16 +1,15 @@
-import React from 'react'
+import React, { useContext } from 'react'
+import { WalletContext } from '../contexts/wallet'
 import { classNames } from '../helpers'
 import Address from './Address'
 
 type addressPillProps = {
   address: string
-  userIsSender: boolean
 }
 
-const AddressPill = ({
-  address,
-  userIsSender,
-}: addressPillProps): JSX.Element => {
+const AddressPill = ({ address }: addressPillProps): JSX.Element => {
+  const { address: walletAddress } = useContext(WalletContext)
```

Comment:
I'm curious why we would want to make the address pill reliant on context. Wouldn't this cause a rerender every time the context had changed, whereas before it would only rerender when the address had changed?

Reply:
So if you check the userIsSender was passed as a prop, which depended on the context, and that would trigger the parent component to re-render and cause all child components to re-render, so I just avoided the prop to be passed and got the context to be used where it is actually required. We were not using the address anywhere else in the parent component